### DISCUSSION TITLE
Dexcom G6 Port 1 Connection Problem

### DISCUSSION THREAD

**Discussion by enginemr2** [NightscoutFoundation/xDrip#2329] [Q&A]
I am using xDrip+ version 2022.05.19
I have enabled the Engineering mode and programmed Port 1 for a connection. This works for 1 or 2 data reports but then stops working, no more data reports. I have discovered that if I have enabled the Event Log screen on my phone the reports will reliably continue. The reports continue for only about 10 hours before terminating. They can be restored by returning to the xDrip Home screen and going back to the Event Log screen. I would like to remain on the Home screen with reliable data reports.

**Comment by enginemr2** [comment]
Actually I am using xDrip version 2022.08.03 to monitor port 1 of the Dexcom G6. I have 2 identical phones to monitor Port 1 and port 2. The phone that is monitoring port 2 is using xDrip version 2021.03.08 since it is also synced to a smartwatch, SWR50. 
When the dual monitoring first states there is some delay between the data from Port 1 to the data from Port 2. The delay varies from almost zero to a couple of seconds. After about 14 hours the delay gets very long more than 10 seconds and then the phone monitoring Port 1 stops responding. The log report " root error". Not sure of the exact error message at this time but will supply if in question.

**Reply by Navid200** [reply]
This is verry interesting and not what I expected.
I expect xDrip to wake up not exactly 5 minutes after it connects.  But, rather earlier.  And as soon as it receives, I expect it to set a timer to wake up not exactly at 5 minutes but sooner.
But, I have never confirmed this is what the code does.  If it does, it will effectively track the timing of the transmitter and gradually move with it.

I will ask someone who knows more to have a look. 
Thanks

**Comment by jamorham** [comment]
Suggest enabling OB1 debugging: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector and then submit logs if the problem is reproducible.
